src/SegmentedString.chpl

```diff
@@ -207,9 +207,7 @@ module SegmentedString {
 
     proc this(const slice: range(strides=strideKind.any)) throws {
       var aa = makeDistArray(slice.size, int);
-      forall (elt,j) in zip(aa, slice) with (var agg = newSrcAggregator(int)) {
-        agg.copy(elt,j);
-      }
+      aa = slice;
```

Comment:
I tried out a `newDstAggregator` instead of the `newSrcAggregator`, and that fixed the bug as well.  I don't know which one is more performant though.

Reply:
oh nice!! I don't think there's any need for aggregation since they share the same domain (I just didn't know what I was doing when I wrote this lol) but I looked at the comm diagnostics and there's nearly no communication from this operation